cpp/include/cudf/join.hpp

```diff
@@ -396,6 +395,39 @@ std::unique_ptr<cudf::table> left_semi_join(
   null_equality compare_nulls         = null_equality::EQUAL,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Returns a vector of bools indicating whether there exist matching row(s) on the right
+ *        table for each row on the left table.
+ *
+ * Given two tables, each row in the `left_keys` table is search for matching rows in the
+ * `right_keys` table. The search result (found or not) is written into the corresponding row of the
+ * output array.
+ *
+ * The resulting vector describe exactly which rows of the left table should be returned from the
+ * query `left_semi_join(left_keys, right_keys, compare_nulls, nan_equality::UNEQUAL)`.
+ *
+ * @code{.pseudo}
+ * left_keys: {{0, 1, 2}}
+ * right_keys: {{5, 4, 1, 2, 3}}
+ * output: {false, true, true}
+ * @endcode
+ *
+ * @param left_keys The left table
+ * @param right_keys The right table
+ * @param compare_nulls Control whether null keys should be considered as matching or not
+ * @param compare_nans Control whether NaN keys in floating-point data should be considered as
+ *        matching or not
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @return A vector of bools indicating if each row in `left_keys` has matching row(s) in
+ *         `right_keys`
+ */
+rmm::device_uvector<bool> left_semi_join_contains(
```

Comment:
I think there may have been a misunderstanding about my suggestion for `contains` and `semi_join` to share an implementation. 

We should distill the common "contains" functionality between `semi_join` and `cudf::contains` to a `detail` API that both of them use.

Reply:
In other words, we shouldn't need to be exposing a new public function with an awkward name like `left_semi_join_contains`.